v3/bind.go

```diff
@@ -442,6 +445,24 @@ func (l *Conn) NTLMBind(domain, username, password string) error {
 	return err
 }
 
+// NTLMUnauthenticatedBind performs an unauthenticated bind.
+//
+// A username may be provided for trace (e.g. logging) purpose only, but it is normally not
```

Comment:
https://winprotocoldoc.blob.core.windows.net/productionwindowsarchives/MS-NLMP/%5BMS-NLMP%5D.pdf
> [...] If the user name and response are empty, the server authenticates the client as the ANONYMOUS user... 

In the same document, see section "3.3.2 NTLM v2 Authentication" the pseudo code for the ServerChallenge:
```
If (User is set to "" && Passwd is set to "") <--
    -- Special case for anonymous authentication
    Set NtChallengeResponseLen to 0
    Set NtChallengeResponseMaxLen to 0
    Set NtChallengeResponseBufferOffset to 0
    Set LmChallengeResponse to Z(1)
[...]
```

I interpret that for a anonymous NTLM bind the username must be empty. I don't have an Active Directory server at hand as I'm on vacation, so I can't confirm the behaviour :/

Can someone confirm the behaviour?

Reply:
Did not notice that paper. The results of my tests are below:
- A username with flag **Password Not Required** and an empty password (or the hash for empty password): A successful logon and I was able to use my ldap connection to retrieve information
- A blank username and password: The error `parsing ntlm-challenge: Anonymous authentication not supported` is returned

Reply:
Thank you for your tests, @nodauf !

It seems anonymous NTLM authentications aren't supported by the Go NTLM library:
```go
//ProcessChallenge crafts an AUTHENTICATE message in response to the CHALLENGE message
//that was received from the server
func ProcessChallenge(challengeMessageData []byte, user, password string) ([]byte, error) {
	if user == "" && password == "" {
		return nil, errors.New("Anonymous authentication not supported")
	}
```
May you update the comment of the function and remove these lines, as atleast a username is required for the NTLM challenge to succeed?
```
// A username may be provided for trace (e.g. logging) purpose only, but it is normally not
// authenticated or otherwise validated by the LDAP server.
```

Reply:
Thank you for taking the time to look. The comments have been updated (in both root folder and v3 :sweat_smile:  )